app/merginapi.cpp

```diff
@@ -504,20 +512,53 @@ void MerginApi::pingMerginReplyFinished()
   emit pingMerginFinished( apiVersion, msg );
 }
 
-void MerginApi::projectDeleted( const QString &projectName )
+ProjectList MerginApi::parseAllProjectsData()
+{
+  QStringList entryList = QDir( mDataDir ).entryList( QDir::NoDotAndDotDot | QDir::Dirs );
+  ProjectList projects;
+
+  for ( QString folderName : entryList )
+  {
+    QFileInfo info( mDataDir + folderName + "/" + mCacheFile );
+    if ( info.exists() )
+    {
+      std::shared_ptr<MerginProject> project = parseProjectData( mDataDir + folderName, mCacheFile );
+      if ( project )
+      {
+        projects << project;
+      }
+    }
+  }
+
+  return projects;
+}
+
+void MerginApi::projectDeleted( const QString &projecFullName )
 {
   for ( std::shared_ptr<MerginProject> project : mMerginProjects )
   {
-    if ( project->name == projectName )
+    if ( getFullProjectName( project->projectNamespace, project->name ) == projecFullName )
     {
       project->status = ProjectStatus::NoVersion;
       project->lastSync = QDateTime();
       project->updated = QDateTime();
+      project->serverUpdated = QDateTime();
+      project->projectDir.clear();
       emit merginProjectsChanged();
     }
   }
 }
 
+void MerginApi::projectDeletedOnPath( const QString &projectDir )
+{
+  for ( std::shared_ptr<MerginProject> project : mMerginProjects )
+  {
+    if ( project->projectDir == mDataDir + projectDir )
+    {
+      projectDeleted( getFullProjectName( project->projectNamespace, project->name ) );
```

Comment:
looks like we have a double for-cycle here going through list of mergin projects :-O

how about we get rid of MerginApi::projectDeleted() and simplify the code?

Reply:
you have added new function getProject() - why not use it in projectDeletedOnPath() ?